wordpress-mobile/WordPress-Android
# Feature/android 14

This is the feature branch for the Android 14 update which contains the following PRs:

* [Set remaining foreground service types](https://github.com/wordpress-mobile/WordPress-Android/pull/21047)
* [Catch ZipException when inflating](https://github.com/wordpress-mobile/WordPress-Android/pull/21060)
* [Android 14 broadcast export behavior](https://github.com/wordpress-mobile/WordPress-Android/pull/21061)

## To Test:

* There's not much to test other than to make sure the app runs as expected on both Android 34+ devices and earlier devices. However, it's important to specifically test site creation and login on an Android 34+ device and verify there are no issues, then do the same on a pre-Android 34 device.

-----

## Notes:

* Even though `LoginWpcomService` is part of our login library, we declare the foreground service type for that service in WPAndroid's manifest so no changes are needed in the service itself (which already targets SDK 34). Also see [this related login PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/150) and [this related WooCommerce PR](https://github.com/woocommerce/woocommerce-android/pull/11970).
* I have not updated the release notes for this PR nor have I assigned a milestone because our milestones are currently inaccurate
* I requested multiple reviewers because I'd like more than one set of eyes on it

-----

## Regression Notes

1. Potential unintended areas of impact

    - Failures in SiteCreationService and LoginWpcomService

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - I have not added tests under the assumption that existing tests would suffice

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
